NebulaPatcher/Patches/Dynamic/UIOptionWindow_Patch.cs

```diff
@@ -0,0 +1,279 @@
+            RectTransform revertButton = multiplayerContent.Find("revert-button").GetComponent<RectTransform>();
+            UIButton[] newRevertButtons = CollectionExtensions.AddToArray(revertButtons, revertButton.GetComponent<UIButton>());
+            AccessTools.Field(__instance.GetType(), "revertButtons").SetValue(__instance, newRevertButtons);
+
+            // Find control templates
+            foreach (RectTransform child in multiplayerContent)
+            {
+                if (child != revertButton)
+                {
+                    Object.Destroy(child.gameObject);
+                }
+            }
+
+            checkboxTemplate = contentTemplate.Find("fullscreen").GetComponent<RectTransform>();
+            comboBoxTemplate = contentTemplate.Find("resolution").GetComponent<RectTransform>();
+            sliderTemplate = contentTemplate.Find("dofblur").GetComponent<RectTransform>();
+
+            inputTemplate = Object.Instantiate(checkboxTemplate, multiplayerContent, false);
+            Object.Destroy(inputTemplate.Find("CheckBox").gameObject);
+            RectTransform inputField = Object.Instantiate(UIRoot.instance.saveGameWindow.transform.Find("input-filename/InputField").GetComponent<RectTransform>(), inputTemplate, false);
+            Vector2 fieldPosition = checkboxTemplate.GetChild(0).GetComponent<RectTransform>().anchoredPosition;
+            inputField.anchoredPosition = new Vector2(fieldPosition.x + 6, fieldPosition.y);
+            inputField.sizeDelta = new Vector2(inputField.sizeDelta.x, 35);
+            inputTemplate.gameObject.SetActive(false);
+
+            AddMultiplayerOptionsProperties(multiplayerContent);
+        }
+
+        [HarmonyPrefix]
+        [HarmonyPatch("_OnOpen")]
+        public static void _OnOpen_Prefix()
+        {
+            tempMultiplayerOptions = (MultiplayerOptions)Config.Options.Clone();
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch("ApplyOptions")]
+        public static void ApplyOptions()
+        {
+            Config.Options = tempMultiplayerOptions;
+            Config.SaveOptions();
+        }
+
+        [HarmonyPrefix]
+        [HarmonyPatch("OnRevertButtonClick")]
+        public static void OnRevertButtonClick_Prefix(int idx)
+        {
+            if (idx == multiplayerTabIndex)
+            {
+                tempMultiplayerOptions = new MultiplayerOptions();
+            }
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch("TempOptionToUI")]
+        public static void TempOptionToUI_Postfix()
+        {
+            List<PropertyInfo> properties = AccessTools.GetDeclaredProperties(typeof(NebulaModel.MultiplayerOptions));
+            foreach (var prop in properties)
+            {
+                if (tempToUICallbacks.ContainsKey(prop.Name))
+                {
+                    tempToUICallbacks[prop.Name]();
+                }
```

Comment:
A bit easier to follow:

```suggestion
                if (tempToUICallbacks.TryGetValue(prop.Name, out var callback))
                {
                    callback();
                }
```